File: sources/scala/05_alloc_churn/main.scala
// task 05 alloc_churn — expected output: 1274991808
// build: scalac -release 17 main.scala    run: scala Main

object Main {
  def main(args: Array[String]): Unit = {
    var total = 0L
    val slots = new Array[Array[Byte]](256)
    var i = 0L
    while (i < 10000000L) {
      val buf = new Array[Byte](64)
      buf(0) = (i % 256L).toByte
      total += (buf(0) & 0xFF).toLong
      slots((i % 256L).toInt) = buf
      i += 1L
    }
    println(total)
  }
}

File: sources/kotlin/05_alloc_churn/main.kt
// task 05 alloc_churn — expected output: 1274991808
// build: kotlinc main.kt -include-runtime -d prog.jar    run: java -jar prog.jar    [native build: kotlinc-native -opt -o prog main.kt    native run: ./prog]

fun main() {
    var total = 0L
    val slots = arrayOfNulls<ByteArray>(256)
    for (i in 0 until 10000000) {
        val buf = ByteArray(64)
        buf[0] = (i % 256).toByte()
        total += (buf[0].toInt() and 0xFF).toLong()
        slots[i % 256] = buf
    }
    println(total)
}
